recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -97,215 +113,223 @@ class GdalConan(ConanFile):
         "with_odbc": False,
+        "with_opencad": False,
+        "with_opencl": True,
         "with_openjpeg": False,
         "with_openssl": False,
         "with_pcre": False,
         "with_pcre2": False,
+        # "with_pdfium": False,
         "with_pg": False,
         "with_png": True,
         "with_podofo": False,
         "with_poppler": False,
-        "with_proj": True,
+        "with_proj": "deprecated",
+        "with_publicdecompwt": False,
         "with_qhull": True,
+        "with_rasterlite2": False,
+        "with_shapelib": True,
+        "with_spatialite": False,
         "with_sqlite3": True,
+        "with_tiledb": False,
         "with_webp": False,
         "with_xerces": False,
         "with_xml2": False,
-        "with_zlib": True,
+        "with_zlib": "deprecated",
         "with_zstd": False,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
+        export_conandata_patches(self)
+        copy(self, "*.cmake",
+             src=os.path.join(self.recipe_folder, "cmake"),
+             dst=os.path.join(self.export_sources_folder, "src", "cmake", "helpers"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if is_apple_os(self):
+            # FIXME: compilation fails on macOS due to difference in ICONV_CONST value in iconv() signature
+            # https://github.com/conan-io/conan-center-index/pull/19298#issuecomment-1840433415
+            # https://github.com/roboticslibrary/libiconv/blob/v1.17/include/iconv.h.in#L82
+            # https://lists.gnu.org/archive/html/bug-gnu-libiconv/2008-07/msg00005.html
+            self.options.with_libiconv = False
+        if Version(self.version) < "3.7":
+            # Latest versions of Arrow are no longer compatible with GDAL 3.5
+            self.options.with_arrow = False
+        if Version(self.version) < "3.8":
+            del self.options.with_libaec
 
     def configure(self):
-        if self.options.with_crypto != "deprecated":
-            self.output.error("with_crypto option is deprecated, use with_openssl instead.")
-
         if self.options.shared:
-            try:
-                del self.options.fPIC
-            except:
-                pass
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("json-c/0.16")
+        self.requires("json-c/0.17")
         self.requires("libgeotiff/1.7.1")
-
+        self.requires("libtiff/4.6.0")
+        self.requires("proj/9.3.0")
```

Comment:
```suggestion
        self.requires("proj/9.3.1")
```
in order to be able to use libgeotiff from #22095